test/org/opensolaris/opengrok/configuration/messages/ProjectMessageTest.java

```diff
@@ -495,4 +497,53 @@ public void testGetRepos() throws Exception {
         out = new String(m.apply(env));
         Assert.assertEquals("Mercurial", out);
     }
+    
+    @Test
+    @ConditionalRun(RepositoryInstalled.GitInstalled.class)
+    public void testSetGet() throws Exception {
+        Assert.assertTrue(env.isHandleHistoryOfRenamedFiles());
+        List<String> projects = new ArrayList<>();
+        projects.add("mercurial");
+        projects.add("git");
+        Message m;
+        
+        // Add the projects.
+        m = new ProjectMessage();
+        m.setText("add");
+        for (String proj : projects) {
+            m.addTag(proj);
+        }
+        m.apply(env);
+        Assert.assertEquals(2, env.getProjectList().size());
+        
+        // Change their property.
+        m = new ProjectMessage();
+        m.setText("set handleRenamedFiles = false");
```

Comment:
can you try doing this test for `= true`? I'm still not conviced that this works for both of the projects and `false` seems to be the default value.

Reply:
Based on the `applyMessage() -> set -> return`

Reply:
I have not committed the fix yet to see that the test is really failing. Coming in a minute..